packages/apps/ability-uniswap-swap/src/lib/ability-helpers/get-uniswap-quote.ts

```diff
@@ -18,112 +19,112 @@ export const getUniswapQuote = async ({
   tokenInAmount: number;
   tokenOutAddress: string;
   tokenOutDecimals: number;
-}): Promise<{
-  bestQuote: ethers.BigNumber;
-  bestFee: number;
-  amountOutMin: ethers.BigNumber;
-  route: SwapRoute;
-}> => {
-  console.log('Getting Uniswap Quote (getUniswapQuote)', {
-    rpcUrl,
-    chainId,
-    tokenInAddress,
-    tokenInDecimals,
-    tokenInAmount,
-    tokenOutAddress,
-    tokenOutDecimals,
-  });
+  recipient: string;
+}): Promise<SwapRoute> {
+  const activeTimeouts = new Set<NodeJS.Timeout>();
+  const realSetTimeout = global.setTimeout;
+  const realClearTimeout = global.clearTimeout;
+
+  global.setTimeout = ((fn: (...args: unknown[]) => void, ms?: number, ...args: unknown[]) => {
+    const id = realSetTimeout(fn, ms as number, ...args);
+    activeTimeouts.add(id);
+    return id;
+  }) as typeof setTimeout;
+
+  global.clearTimeout = ((id: NodeJS.Timeout) => {
+    activeTimeouts.delete(id);
+    return realClearTimeout(id);
+  }) as typeof clearTimeout;
 
   const provider = new ethers.providers.StaticJsonRpcProvider(rpcUrl);
   const router = new AlphaRouter({ chainId, provider });
+  try {
+    const tokenIn = new Token(chainId, tokenInAddress, tokenInDecimals);
+    const tokenOut = new Token(chainId, tokenOutAddress, tokenOutDecimals);
+    const amountIn = CurrencyAmount.fromRawAmount(
+      tokenIn,
+      ethers.utils.parseUnits(tokenInAmount.toString(), tokenInDecimals).toString(),
+    );
+    const slippage = new Percent(50, 10_000); // 0.5%
```

Comment:
Should we make slippage an input param controlled by the Delegatee?

Reply:
Added in [this commit](https://github.com/LIT-Protocol/Vincent/pull/286/commits/da92d7400c0d87ddde02a92c19d7e3d8518123d0)